### PR TITLE
Remove formatting module

### PR DESCRIPTION
**Pull request by arturbosch**

### CODE REVIEW

File: detekt-formatting/src/main/kotlin/io/gitlab/arturbosch/detekt/formatting/ConsecutiveBlankLines.kt [hunk 1]
```diff
@@ -1,32 +0,0 @@
...
-import io.gitlab.arturbosch.detekt.api.Issue
-import io.gitlab.arturbosch.detekt.api.Severity
-import io.gitlab.arturbosch.detekt.api.TokenRule
-import org.jetbrains.kotlin.com.intellij.psi.PsiWhiteSpace
-import org.jetbrains.kotlin.com.intellij.psi.impl.source.tree.LeafPsiElement
-
-/**
- * @author Artur Bosch
- */
-class ConsecutiveBlankLines(config: Config) : TokenRule(config) {
```
Comment (vanniktech) [inline detekt-formatting/src/main/kotlin/io/gitlab/arturbosch/detekt/formatting/ConsecutiveBlankLines.kt]: I'd like to have this in the normal detekt with a configurable amount? Basically I want to forbid 2 new consecutive new lines.

Comment (arturbosch) [reply detekt-formatting/src/main/kotlin/io/gitlab/arturbosch/detekt/formatting/ConsecutiveBlankLines.kt]: As jetbrains wants to bring their own formatting tool, I don't know how much formatting we should add, if any. On the other side this rule is sometimes broken by contributors ...

**Comment by arturbosch** [comment]
UnusedImports would also be great for the standard rule set.

**Comment by vanniktech** [comment]
UnusedImports is covered by ktlint. And on second thought some ConsecutiveLines is so too.

**Comment by arturbosch** [comment]
This should also handle #390 .

**Comment by arturbosch** [comment]
Code which handles `autoCorrect`-functionality should be removed too.

**Comment by arturbosch** [comment]
Migrate ExpressionBodySyntax rule.

**Comment by vgaidarji** [comment]
Does it mean that work in this PR will be finished, `formatting` module will be removed completely and `detekt` will not support formatting rules anymore?

And from https://github.com/arturbosch/detekt/issues/326 I see 
> let ktlint handle formatting

Does it mean that I should use `detekt` (complexity/code-smell/etc.) + `ktlint` (formatting) to achieve broader rules coverage?

**Comment by arturbosch** [comment]
To make it short: Yes.

The long explanation: It seemed like a good idea to combine every aspect (complexity/codesmell/formatting) into detekt to have everything in one place. The downside is that this requires much more maintenace for the detekt contributors and of course is a bit redundant work with KtLint. 
We get feedback on two sides, the rules diverge, new rules get added. There are also two different paradigm to write rules: standard rules uses Expressions/Statements/Declarations and formatting rules use a low level token approach (complexer).
The last point is that I do not want to deal with users code anymore (false positives lead to more work for the user).
Everything in one plugin (detekt) still sounds great, so we try to provide wrappers over `intellij commandline scripts` (detektIdeaFormat/detektIdeaInspect), `KtLint` (should replace the formatting rule set), 'official jetbrains kotlin formatting tool' (if possible) which will be released soon.

**Comment by arturbosch** [comment]
Merged manually.

**Pull request closed without merging**